File: TrainingSDKBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TrainingSDKBridge, NSObject)

RCT_EXTERN_METHOD(start:(NSString *)user clientId:(NSString *)clientId clientSecret:(NSString *)clientSecret resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(startSession:(NSString *)clientId clientSecret:(NSString *)clientSecret)

RCT_EXTERN_METHOD(logout)

RCT_EXTERN_METHOD(showNutritionModule)

RCT_EXTERN_METHOD(setMainColor:(NSString *)color)

RCT_EXTERN_METHOD(setNavigationBarColor:(NSString *)color)

RCT_EXTERN_METHOD(setNavigationTintColor:(NSString *)color)

RCT_EXTERN_METHOD(showTrainingModule)

RCT_EXTERN_METHOD(setNavigationTitle:(NSString *)title)

RCT_EXTERN_METHOD(updateProfile:(NSString *)sex height:(nonnull NSNumber *)height weight:(nonnull NSNumber *)weight birthDateString:(NSString *)birthDateString resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

@end

File: TrainingSDKBridge.swift
//
//  TrainingSDKBridge.swift
//  SDKTrainingExample
//
//

import Foundation
import TrainingSDK
import React

@objc(TrainingSDKBridge)
class TrainingSDKBridge: NSObject {
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func start(_ user: String, clientId: String, clientSecret: String, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    WorkoutSDK.sharedInstance.start(with: user, clientId: clientId, clientSecret: clientSecret)
    resolve(nil)
  }
  
  @objc
  func updateProfile(sex: String?, height: NSNumber?, weight: NSNumber?, birthDateString: String?, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
      let dateFormatter = DateFormatter()
      dateFormatter.dateFormat = "yyyy-MM-dd" // Asegúrate de que este formato coincida con el que envías desde JS

      let birthDate = birthDateString != nil ? dateFormatter.date(from: birthDateString!) : nil
      let gender = sex != nil ? TrainingGender(rawValue: sex!) : nil

      WorkoutSDK.sharedInstance.updateProfile(sex: gender, height: height?.intValue, weigth: weight?.intValue, birthDate: birthDate)

      resolve(nil)
  }

  @objc
  func showTrainingModule() {
    WorkoutSDK.sharedInstance.showWorkoutPlan()
  }
  
  @objc func startSession(user: String, _ clientId: String, clientSecret: String) {
    WorkoutSDK.sharedInstance.start(with: user, clientId: clientId, clientSecret: clientSecret)
  }
  
  @objc func logout() {
    WorkoutSDK.sharedInstance.logout()
  }
  
  @objc func setMainColor(_ color: String) {
    WorkoutSDK.sharedInstance.setMainColor(color: color)
  }
  
  @objc
  func setNavigationBarColor(_ color: String) {
    WorkoutSDK.sharedInstance.setNavigationBarColor(color: color)
  }

  @objc
  func setNavigationTintColor(_ color: String) {
    WorkoutSDK.sharedInstance.setNavigationTintColor(color: color)
  }

  
}
